collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java

```diff
@@ -170,7 +178,42 @@ public boolean checkAccountPermission() {
     }
 
     public String getSelectedAccount() {
-        return (String) preferences.get(PreferenceKeys.KEY_SELECTED_GOOGLE_ACCOUNT);
+        Account[] googleAccounts = credential.getAllAccounts();
+        String account = (String) preferences.get(PreferenceKeys.KEY_SELECTED_GOOGLE_ACCOUNT);
```

Comment:
This can be moved within the `if-block` which asserts that `googleAccounts` are available